### PR TITLE
Fetch project if saving a new feed

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Previous changes introduced a bug where if creating and saving a new feed source in a project, the project would not be fetched again, caused a loading error. This PR resolves that, which also resolves the current e2e issues.

### CODE REVIEW

**Review by miles-grant-ibigroup — APPROVED** [review]
Thanks for fixing this!

File: lib/manager/actions/projects.js [hunk 1]
```diff
@@ -112,7 +112,8 @@ export function fetchProjectWithFeeds (projectId: string, unsecure: ?boolean) {
     // Don't fetch project or feeds if project and feeds are already present
```
Comment (binh-dam-ibigroup) [inline lib/manager/actions/projects.js]: Can you update this comment?

File: lib/manager/actions/projects.js [hunk 2]
```diff
@@ -112,7 +112,8 @@ export function fetchProjectWithFeeds (projectId: string, unsecure: ?boolean) {
...
     const state = getState()
     const project = state.projects.all.find(p => p.id === projectId)
-    if (project && project.feedSources && project.feedSourceCount && project.feedSources.length === project.feedSourceCount) {
+    const isSaving = state.status.message && state.status.message.includes('Saving')
```
Comment (miles-grant-ibigroup) [inline lib/manager/actions/projects.js]: I just realized this doesn't support i18n, so this fix won't work in Polish or German. Maybe we should set another redux variable when saving?

Comment (binh-dam-ibigroup) [reply lib/manager/actions/projects.js]: Indeed. That would also clarify which action this behavior should apply.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
What Miles said.

File: lib/manager/reducers/status.js [hunk 3]
```diff
@@ -71,7 +72,7 @@ const config = (state: StatusState = defaultState, action: Action): StatusState
     case 'UPDATING_SERVER':
     case 'FETCHING_ALL_JOBS':
     case 'REQUEST_RTD_ALERTS':
-      return {...state, message: messages(action.type)}
+      return {...state, message: messages(action.type), isSaving: action.type === 'SAVING_FEEDSOURCE'}
```
Comment (binh-dam-ibigroup) [inline lib/manager/reducers/status.js]: Is that `isSaving` status getting unset after the the feed source is saved, or through another action above (and which one(s))?

Comment (philip-cline) [reply lib/manager/reducers/status.js]: The parameter will get set to false after the feed source is saved by any subsequent action in the case list. 
image

File: lib/manager/actions/projects.js (hunk 1 above)
Comment (philip-cline) [reply lib/manager/actions/projects.js]: This is updated in [c8f54e8](https://github.com/ibi-group/datatools-ui/pull/901/commits/c8f54e8d224542f56e53b71eb69dbeda954af84a)

File: lib/manager/reducers/status.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/manager/reducers/status.js]: What's setting it back to false? Do you think this could be the source of the ci failures

Comment (binh-dam-ibigroup) [reply lib/manager/reducers/status.js]: From looking at the code, add to the comment on line 75 that `REQUESTING_FEED_SOURCE` and `REQUESTING_PROJECT` set the `isSaving` state back to false, which is what we want.

**Review by binh-dam-ibigroup — APPROVED** [review]
Add a comment regarding resetting the `isSaving` state per https://github.com/ibi-group/datatools-ui/pull/901#discussion_r1093461538.

File: lib/manager/reducers/status.js (hunk 3 above)
Comment (philip-cline) [reply lib/manager/reducers/status.js]: Updated in [ba34342](https://github.com/ibi-group/datatools-ui/pull/901/commits/ba343422d596b78b3aa21de98f7fc734ca288ff4)

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Looks like the tests are failing randomly -- I'd remove the extra `wait`s and instead try to fix the problem those waits are solving.

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks like things are working now!

**Comment by philip-cline** [comment]
This has gone back and forth a bit since your last review @binh-dam-ibigroup, so just requesting another review

File: lib/manager/actions/projects.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/manager/actions/projects.js]: Is the update to the comment you made gone?

File: lib/manager/reducers/status.js [hunk 4]
```diff
@@ -71,7 +73,7 @@ const config = (state: StatusState = defaultState, action: Action): StatusState
     case 'UPDATING_SERVER':
     case 'FETCHING_ALL_JOBS':
     case 'REQUEST_RTD_ALERTS':
-      return {...state, message: messages(action.type)}
+      return {...state, saving: true, message: messages(action.type)}
```
Comment (binh-dam-ibigroup) [inline lib/manager/reducers/status.js:76]: Add comment regarding what sets `saving` back to `false`. Also, a previous commit was setting `saving` back to `false` but I'm not seeing that, meaning the logic is incomplete.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Looks strange. Something happened and some previous valid changes were lost.

**Comment by miles-grant-ibigroup** [comment]
@binh-dam-ibigroup all changes should be back now!

**Comment by binh-dam-ibigroup** [comment]
> @binh-dam-ibigroup all changes should be back now!

Great, thanks!

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**